pkg/bpf/map.go

```diff
@@ -147,7 +147,6 @@ func NewMap(name string, mapType MapType, keySize int, valueSize int, maxEntries
 		name:       path.Base(name),
 		dumpParser: dumpParser,
 	}
-	m.setPathIfUnset()
```

Comment:
I don't know this code but was this intentional? it seems a little random

Reply:
Yes, removing the code from there was intentional. `NewMap` function is called in `init()` functions inside `pkg/map` package. Setting the path already requires the `mapRoot`, Of course `init` functions are called *before* Cilium even decides what `mapRoot` should look like. That little bug is kinda invisible when Cilium is using `/sys/fs/bpf` by default.

The only way to reproduce that problem on the current master is to define a custom `mapRoot` via cli arguments to cilium-agent and we were "lucky" that no one complained about it. But after implementing the logic about fallback to `/run/cilium/bpffs` is becomes a serious problem - `init` functions are trying to pick `/sys/fs/bpf` despite the fact that the other path is going to be chosen.

Reply:
Removing that like is just fine, because `setPathIfUnset` method is called also in `Open` and `OpenOrCreate` methods:

https://github.com/cilium/cilium/blob/master/pkg/bpf/map.go#L265
https://github.com/cilium/cilium/blob/master/pkg/bpf/map.go#L302

And those are the proper places, because `mapRoot` is already correct when we are opening and creating maps explicitly.